sbaid/model/results/cross_section_snapshot.py

```diff
@@ -52,10 +52,24 @@ def __init__(self, snapshot_id: str, cross_section_snapshot_id: str, cross_secti
         super().__init__(snapshot_id=snapshot_id,
                          cross_section_snapshot_id=cross_section_snapshot_id,
                          cross_section_name=cross_section_name,
-                         b_display=b_display)
+                         b_display=b_display,
+                         lane_snapshots=Gio.ListStore.new(LaneSnapshot))
 
     def load_from_db(self) -> None:
         """todo"""
 
     def add_lane_snapshot(self, snapshot: LaneSnapshot) -> None:
-        """todo"""
+        """Add a LaneSnapshot to this CrossSectionSnapshot."""
+        self.lane_snapshots.append(snapshot)
+
+    def calculate_cs_average_speed(self) -> float:
+        """Calculate the average of the average speed value from all lanes in this cross section snapshot. """
+        counter = 0
+        n = len(self.lane_snapshots)
+        for i in range(n):
+            lane_snapshot = self.lane_snapshots.get_item(i)
+            counter += lane_snapshot.average_speed
+
+        return counter / n
```

Comment:
sexy one-liner that does the same thing
```suggestion
        return sum([lane_snapshot.average_speed for lane_snapshot in self.lane_snapshots]) / len(self.lane_snapshots)
```

Reply:
did you know you can comment stuff normally without calling them sexy?